src/details/ArborX_DetailsHappyTreeFriends.hpp

```diff
@@ -52,20 +52,19 @@ struct HappyTreeFriends
   }
 
   template <class BVH>
-  static KOKKOS_FUNCTION decltype(auto) getBoundingVolume(BVH const &bvh, int i)
+  static KOKKOS_FUNCTION auto const &getBoundingVolume(BVH const &bvh, int i)
   {
     return bvh._internal_and_leaf_nodes(i).bounding_volume;
   }
 
   template <class BVH>
-  static KOKKOS_FUNCTION decltype(auto) isLeaf(BVH const &bvh, int i)
+  static KOKKOS_FUNCTION bool isLeaf(BVH const &bvh, int i)
   {
-    return i >= static_cast<int>(bvh.size()) - 1;
+    return bvh._internal_and_leaf_nodes(i).isLeaf();
```

Comment:
Did you intentionally push this change? I know we discussed it, but it has nothing to do with return types.

Reply:
I did but I don't mind reverting if you want to.